resources/qml/Settings/SettingView.qml

```diff
@@ -9,229 +9,406 @@ import QtQuick.Layouts 1.1
+
+            placeholderText: catalog.i18nc("@label:textbox", "Filter...")
+
+            style: TextFieldStyle
             {
-                switch(model.type)
-                {
-                    case "int":
-                        return "SettingTextField.qml"
-                    case "float":
-                        return "SettingTextField.qml"
-                    case "enum":
-                        return "SettingComboBox.qml"
-                    case "extruder":
-                        return "SettingExtruder.qml"
-                    case "bool":
-                        return "SettingCheckBox.qml"
-                    case "str":
-                        return "SettingTextField.qml"
-                    case "category":
-                        return "SettingCategory.qml"
-                    default:
-                        return "SettingUnknown.qml"
-                }
+                textColor: UM.Theme.getColor("setting_control_text");
+                font: UM.Theme.getFont("default");
+                background: Item {}
             }
 
-            // Binding to ensure that the right containerstack ID is set for the provider.
-            // This ensures that if a setting has a limit_to_extruder id (for instance; Support speed points to the
-            // extruder that actually prints the support, as that is the setting we need to use to calculate the value)
-            Binding
+            property var expandedCategories
+            property bool lastFindingSettings: false
+
+            onTextChanged:
             {
-                target: provider
-                property: "containerStackId"
-                when: model.settable_per_extruder || (inheritStackProvider.properties.limit_to_extruder != null && inheritStackProvider.properties.limit_to_extruder >= 0);
-                value:
+                definitionsModel.filter = {"label": "*" + text};
+                findingSettings = (text.length > 0);
+                if(findingSettings != lastFindingSettings)
                 {
-                    if(!model.settable_per_extruder || machineExtruderCount.properties.value == 1)
-                    {
-                        //Not settable per extruder or there only is global, so we must pick global.
-                        return Cura.MachineManager.activeMachineId;
-                    }
-                    if(inheritStackProvider.properties.limit_to_extruder != null && inheritStackProvider.properties.limit_to_extruder >= 0)
+                    if(findingSettings)
                     {
-                        //We have limit_to_extruder, so pick that stack.
-                        return ExtruderManager.extruderIds[String(inheritStackProvider.properties.limit_to_extruder)];
+                        expandedCategories = definitionsModel.expanded.slice();
+                        definitionsModel.expanded = ["*"];
+                        definitionsModel.showAncestors = true;
```

Comment:
Won't this also show normal parents, such as "Line Width" is a parent of "Wall Line Width"? Is that intended?